mico-core/src/main/java/io/github/ust/mico/core/web/ApplicationController.java

```diff
@@ -103,13 +103,18 @@
     }
 
     @PostMapping
-    public ResponseEntity<Resource<MicoApplication>> createApplication(@RequestBody MicoApplication newApplication) {
+    public ResponseEntity<?> createApplication(@Valid @RequestBody MicoApplication newApplication,
+                                               Errors validationErrors) {
         Optional<MicoApplication> applicationOptional = applicationRepository.
             findByShortNameAndVersion(newApplication.getShortName(), newApplication.getVersion());
         if (applicationOptional.isPresent()) {
             throw new ResponseStatusException(HttpStatus.CONFLICT,
                 "Application '" + newApplication.getShortName() + "' '" + newApplication.getVersion() + "' already exists.");
         }
+        micoDataValidator.validateMicoApplication(newApplication, validationErrors);
+        if (validationErrors.hasErrors()) {
+            return new ResponseEntity<>(new ValidationErrorDTO(validationErrors), ValidationErrorDTO.HTTP_STATUS);
```

Comment:
Why don't you use **throw new Exception**?

Reply:
I want to return the resource `ValidationErrorDTO` as the body of the response and not only a string. That's the reason why I can't throw an exception here. Or do you know a way how to accomplish that by throwing an exception?